### PR TITLE
add logplex_ranch_metrics

### PR DESCRIPTION
**Pull request by danp**
Emits connection counts for logplex_logs_rest and logplex_api_v3.

Mostly copied from #220

### CODE REVIEW

File: src/logplex_ranch_metrics.erl [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+     logplex_logs_rest,
+     logplex_api_v3,
+    ].
+
+notify_metric(Server) ->
+    Metric = metric(Server),
+    Value = value(Server),
+    folsom_metrics:notify(Metric, Value, gauge).
+
+metric(CtrlRod) ->
```
Comment (ypaq) [inline src/logplex_ranch_metrics.erl]: s/CtrlRod/Server/

File: src/logplex_ranch_metrics.erl [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+%% @private
+handle_call(Msg, _From, State) ->
+    ?WARN("type=unexpected_call msg='~p'", [Msg]),
+    Error = {unexpected_call, Msg},
+    {reply, Error, State}.
+
+%% @private
+handle_cast(report_status, State) ->
+    [ notify_metric(Server) || Server <- servers()],
```
Comment (ypaq) [inline src/logplex_ranch_metrics.erl]: s/servers/ranch_servers

File: src/logplex_ranch_metrics.erl [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+refresh_cycle() ->
+    Timeout = timer:seconds(60),
+    erlang:send_after(Timeout, self(), cycle_timeout),
+    ok.
+
+ranch_servers() ->
+    [
+     logplex_logs_rest,
+     logplex_api_v3,
```
Comment (ypaq) [inline src/logplex_ranch_metrics.erl]: Remove `,`.

**Comment by danp** [comment]
Updated, PTAL.

File: rebar.lock [hunk 4]
```diff
@@ -22,6 +22,10 @@
   {git,"git://github.com/ninenines/cowlib.git",
        {ref,"d544a494af4dbc810fc9c15eaf5cc050cced1501"}},
   1},
+ {<<"erlang_syslog">>,
+  {git,"https://github.com/heroku/erlang_syslog",
+       {ref,"0cfef5a95150e11f60fee5c35b24decfcc41066e"}},
+  0},
```
Comment (ypaq) [inline rebar.lock]: This seems wrong. When you look at https://github.com/heroku/erlang_syslog/blob/0cfef5a95150e11f60fee5c35b24decfcc41066e/src/syslog.app.src#L1 the app is called `syslog`. The commit seems correct. I'm not sure why rebar does this.

**Comment by danp** [comment]
Would like to add emission of what the current connection limit is for each server, using [ranch_server:get_max_connections](https://github.com/ninenines/ranch/blob/1.4.0/src/ranch_server.erl#L93-L95).

**Comment by danp** [comment]
@ypaq updated this and CI is green now. PTAL!

**Pull request merged**